android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -338,31 +356,30 @@ public static boolean hasConnection(Context context) {
   private static void copyAssets(Context context) {
     AssetManager assetManager = context.getAssets();
     try {
-      File file;
       // Copy main files
       copyAsset(context, KMFilename_KeyboardHtml, "", true);
       copyAsset(context, KMFilename_JSEngine, "", true);
       copyAsset(context, KMFilename_KmwCss, "", true);
       copyAsset(context, KMFilename_Osk_Ttf_Font, "", true);
       copyAsset(context, KMFilename_Osk_Woff_Font, "", true);
 
-      // Copy default keyboard assets
-      String packagesDir = context.getDir("data", Context.MODE_PRIVATE) + File.separator + KMDefault_AssetPackages;
-      file = new File(packagesDir);
-      if (!file.exists()) {
-        file.mkdir();
+      String packagesDirStr = context.getDir("data", Context.MODE_PRIVATE).toString() +
```

Comment:
It'd be nice to define helper functions for all these folder concatenation operations. e.g. `getPackagesDir()` or similar, so all path construction is in a single place for consistency and elimination of bugs.